dist/template: fix YAML indentation

lucab opened the pull request:
This fixes several YAML sections that are over or under indented.
It contains no functional changes.

/cc @steveeJ

steveeJ commented:
Which tool did you use to autoformat this?

lucab commented:
@steveeJ nothing for now. I just spot a couple of sections off, but the few linters/formatters I tried were making a larger mess, so I just adjusted them manually. I'll report back if/when I find something that I like, to hook in CI.

steveeJ commented:
> I'll report back if/when I find something that I like, to hook in CI.

Thanks, that sounds like a reasonable goal. Let's keep it open until then.

lucab commented:
I have added a linting script and fixed up all errors, PTAL. @steveeJ can you try if the script works for you? If so, once this lands I'll wire it up as a prow job.

steveeJ commented on dist/prow_yaml_lint.sh:
```diff
@@ -0,0 +1,14 @@
+  exit 1
+fi
+
+find . -type f -name '*.yaml' -print0 | xargs -0 "${YAML_LINT_CMD[@]}"
```
I think we could also match `*.yml` here

lucab commented:
I've added the `.yml` portion too, but the green-stamp is now gone. It needs another round of review.

steveeJ commented:
/lgtm

lucab commented:
/hold cancel
